File: tests/mocks/mock_services.go
// Package mocks provides mock implementations for testing SIEM/SOAR services
package mocks

import (
	"context"
	"encoding/json"
	"fmt"
	"net/http"
	"net/http/httptest"
	"sync"
	"time"
)

// MockAlert represents a mock alert
type MockAlert struct {
	ID          string                 `json:"id"`
	Title       string                 `json:"title"`
	Severity    string                 `json:"severity"`
	Status      string                 `json:"status"`
	Source      string                 `json:"source"`
	Timestamp   time.Time              `json:"timestamp"`
	Description string                 `json:"description"`
	Metadata    map[string]interface{} `json:"metadata"`
}

// MockEvent represents a mock event
type MockEvent struct {
	EventID   string                 `json:"event_id"`
	Timestamp time.Time              `json:"timestamp"`
	EventType string                 `json:"event_type"`
	Source    map[string]interface{} `json:"source"`
	Message   string                 `json:"message"`
	Metadata  map[string]interface{} `json:"metadata"`
}

// MockRule represents a mock detection rule
type MockRule struct {
	ID          string `json:"id"`
	Name        string `json:"name"`
	Description string `json:"description"`
	Severity    string `json:"severity"`
	Query       string `json:"query"`
	Enabled     bool   `json:"enabled"`
}

// MockPlaybook represents a mock playbook
type MockPlaybook struct {
	ID          string      `json:"id"`
	Name        string      `json:"name"`
	Description string      `json:"description"`
	Enabled     bool        `json:"enabled"`
	Steps       []MockStep  `json:"steps"`
}

// MockStep represents a mock playbook step
type MockStep struct {
	ID     string `json:"id"`
	Name   string `json:"name"`
	Type   string `json:"type"`
	Action string `json:"action"`
}

// MockDataStore provides in-memory storage for mock data
type MockDataStore struct {
	mu        sync.RWMutex
	alerts    map[string]MockAlert
	events    []MockEvent
	rules     map[string]MockRule
	playbooks map[string]MockPlaybook
}

// NewMockDataStore creates a new mock data store with sample data
func NewMockDataStore() *MockDataStore {
	store := &MockDataStore{
		alerts:    make(map[string]MockAlert),
		events:    []MockEvent{},
		rules:     make(map[string]MockRule),
		playbooks: make(map[string]MockPlaybook),
	}
	store.loadSampleData()
	return store
}

func (s *MockDataStore) loadSampleData() {
	// Sample alerts
	s.alerts["alert-001"] = MockAlert{
		ID:          "alert-001",
		Title:       "SSH Brute Force Detected",
		Severity:    "high",
		Status:      "new",
		Source:      "detection_engine",
		Timestamp:   time.Now().Add(-1 * time.Hour),
		Description: "Multiple failed SSH login attempts detected",
	}

	s.alerts["alert-002"] = MockAlert{
		ID:          "alert-002",
		Title:       "Malware Detected",
		Severity:    "critical",
		Status:      "investigating",
		Source:      "edr",
		Timestamp:   time.Now().Add(-30 * time.Minute),
		Description: "Ransomware signature detected on endpoint",
	}

	// Sample rules
	s.rules["rule-001"] = MockRule{
		ID:          "rule-001",
		Name:        "SSH Brute Force Detection",
		Description: "Detects multiple failed SSH login attempts",
		Severity:    "high",
		Query:       "event_type = 'auth_failure' AND service = 'sshd' | count() > 5 by src_ip window 5m",
		Enabled:     true,
	}

	// Sample playbooks
	s.playbooks["playbook-001"] = MockPlaybook{
		ID:          "playbook-001",
		Name:        "SSH Brute Force Response",
		Description: "Automated response to SSH brute force attacks",
		Enabled:     true,
		Steps: []MockStep{
			{ID: "step-1", Name: "Enrich IP", Type: "action", Action: "enrichment.ip_lookup"},
			{ID: "step-2", Name: "Block IP", Type: "action", Action: "firewall.block_ip"},
		},
	}
}

// MockGatewayServer creates a mock API gateway server
type MockGatewayServer struct {
	*httptest.Server
	store *MockDataStore
}

// NewMockGatewayServer creates a new mock gateway server
func NewMockGatewayServer() *MockGatewayServer {
	store := NewMockDataStore()

	mux := http.NewServeMux()

	// Health endpoints
	mux.HandleFunc("/health", func(w http.ResponseWriter, r *http.Request) {
		w.Header().Set("Content-Type", "application/json")
		json.NewEncoder(w).Encode(map[string]string{
			"status":  "healthy",
			"service": "gateway",
		})
	})

	mux.HandleFunc("/ready", func(w http.ResponseWriter, r *http.Request) {
		w.Header().Set("Content-Type", "application/json")
		json.NewEncoder(w).Encode(map[string]string{
			"status": "ready",
		})
	})

	// Alert endpoints
	mux.HandleFunc("/api/v1/alerts", func(w http.ResponseWriter, r *http.Request) {
		w.Header().Set("Content-Type", "application/json")
		switch r.Method {
		case http.MethodGet:
			store.mu.RLock()
			alerts := make([]MockAlert, 0, len(store.alerts))
			for _, alert := range store.alerts {
				alerts = append(alerts, alert)
			}
			store.mu.RUnlock()
			json.NewEncoder(w).Encode(map[string]interface{}{
				"alerts": alerts,
				"total":  len(alerts),
			})
		case http.MethodPost:
			var alert MockAlert
			if err := json.NewDecoder(r.Body).Decode(&alert); err != nil {
				http.Error(w, err.Error(), http.StatusBadRequest)
				return
			}
			store.mu.Lock()
			alert.ID = fmt.Sprintf("alert-%d", time.Now().UnixNano())
			alert.Timestamp = time.Now()
			store.alerts[alert.ID] = alert
			store.mu.Unlock()
			w.WriteHeader(http.StatusCreated)
			json.NewEncoder(w).Encode(alert)
		default:
			http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
		}
	})

	// Rules endpoints
	mux.HandleFunc("/api/v1/rules", func(w http.ResponseWriter, r *http.Request) {
		w.Header().Set("Content-Type", "application/json")
		switch r.Method {
		case http.MethodGet:
			store.mu.RLock()
			rules := make([]MockRule, 0, len(store.rules))
			for _, rule := range store.rules {
				rules = append(rules, rule)
			}
			store.mu.RUnlock()
			json.NewEncoder(w).Encode(map[string]interface{}{
				"rules": rules,
			})
		case http.MethodPost:
			var rule MockRule
			if err := json.NewDecoder(r.Body).Decode(&rule); err != nil {
				http.Error(w, err.Error(), http.StatusBadRequest)
				return
			}
			store.mu.Lock()
			rule.ID = fmt.Sprintf("rule-%d", time.Now().UnixNano())
			store.rules[rule.ID] = rule
			store.mu.Unlock()
			w.WriteHeader(http.StatusCreated)
			json.NewEncoder(w).Encode(map[string]string{"message": "rule created"})
		default:
			http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
		}
	})

	// Playbook endpoints
	mux.HandleFunc("/api/v1/playbooks", func(w http.ResponseWriter, r *http.Request) {
		w.Header().Set("Content-Type", "application/json")
		switch r.Method {
		case http.MethodGet:
			store.mu.RLock()
			playbooks := make([]MockPlaybook, 0, len(store.playbooks))
			for _, pb := range store.playbooks {
				playbooks = append(playbooks, pb)
			}
			store.mu.RUnlock()
			json.NewEncoder(w).Encode(map[string]interface{}{
				"playbooks": playbooks,
			})
		case http.MethodPost:
			w.WriteHeader(http.StatusCreated)
			json.NewEncoder(w).Encode(map[string]string{"message": "playbook created"})
		default:
			http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
		}
	})

	// Query endpoint
	mux.HandleFunc("/api/v1/query", func(w http.ResponseWriter, r *http.Request) {
		w.Header().Set("Content-Type", "application/json")
		if r.Method != http.MethodPost {
			http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
			return
		}
		json.NewEncoder(w).Encode(map[string]interface{}{
			"results":  []interface{}{},
			"metadata": map[string]interface{}{"total": 0, "duration_ms": 10},
		})
	})

	server := httptest.NewServer(mux)

	return &MockGatewayServer{
		Server: server,
		store:  store,
	}
}

// MockCollectorServer creates a mock event collector server
type MockCollectorServer struct {
	*httptest.Server
	events []MockEvent
	mu     sync.RWMutex
}

// NewMockCollectorServer creates a new mock collector server
func NewMockCollectorServer() *MockCollectorServer {
	collector := &MockCollectorServer{
		events: []MockEvent{},
	}

	mux := http.NewServeMux()

	mux.HandleFunc("/health", func(w http.ResponseWriter, r *http.Request) {
		w.Header().Set("Content-Type", "application/json")
		json.NewEncoder(w).Encode(map[string]string{
			"status":  "healthy",
			"service": "collector",
		})
	})

	mux.HandleFunc("/api/v1/events", func(w http.ResponseWriter, r *http.Request) {
		w.Header().Set("Content-Type", "application/json")
		if r.Method != http.MethodPost {
			http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
			return
		}

		var payload struct {
			Events []MockEvent `json:"events"`
		}
		if err := json.NewDecoder(r.Body).Decode(&payload); err != nil {
			http.Error(w, err.Error(), http.StatusBadRequest)
			return
		}

		collector.mu.Lock()
		collector.events = append(collector.events, payload.Events...)
		collector.mu.Unlock()

		w.WriteHeader(http.StatusAccepted)
		json.NewEncoder(w).Encode(map[string]interface{}{
			"accepted": len(payload.Events),
		})
	})

	mux.HandleFunc("/api/v1/events/batch", func(w http.ResponseWriter, r *http.Request) {
		w.Header().Set("Content-Type", "application/json")
		if r.Method != http.MethodPost {
			http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
			return
		}

		var payload struct {
			Events []MockEvent `json:"events"`
		}
		if err := json.NewDecoder(r.Body).Decode(&payload); err != nil {
			http.Error(w, err.Error(), http.StatusBadRequest)
			return
		}

		collector.mu.Lock()
		collector.events = append(collector.events, payload.Events...)
		collector.mu.Unlock()

		json.NewEncoder(w).Encode(map[string]interface{}{
			"accepted": len(payload.Events),
		})
	})

	collector.Server = httptest.NewServer(mux)
	return collector
}

// GetEvents returns collected events
func (c *MockCollectorServer) GetEvents() []MockEvent {
	c.mu.RLock()
	defer c.mu.RUnlock()
	return append([]MockEvent{}, c.events...)
}

// MockClickHouseClient provides a mock ClickHouse client
type MockClickHouseClient struct {
	mu      sync.RWMutex
	queries []string
	results map[string]interface{}
}

// NewMockClickHouseClient creates a new mock ClickHouse client
func NewMockClickHouseClient() *MockClickHouseClient {
	return &MockClickHouseClient{
		queries: []string{},
		results: map[string]interface{}{
			"SELECT count() FROM events": []map[string]interface{}{{"count()": 1000}},
		},
	}
}

// Query executes a mock query
func (c *MockClickHouseClient) Query(ctx context.Context, query string) (interface{}, error) {
	c.mu.Lock()
	c.queries = append(c.queries, query)
	c.mu.Unlock()

	c.mu.RLock()
	result, ok := c.results[query]
	c.mu.RUnlock()

	if ok {
		return result, nil
	}
	return []map[string]interface{}{}, nil
}

// GetQueries returns executed queries
func (c *MockClickHouseClient) GetQueries() []string {
	c.mu.RLock()
	defer c.mu.RUnlock()
	return append([]string{}, c.queries...)
}

// MockKafkaProducer provides a mock Kafka producer
type MockKafkaProducer struct {
	mu       sync.RWMutex
	messages []struct {
		Topic string
		Key   string
		Value []byte
	}
}

// NewMockKafkaProducer creates a new mock Kafka producer
func NewMockKafkaProducer() *MockKafkaProducer {
	return &MockKafkaProducer{}
}

// Produce sends a message to the mock producer
func (p *MockKafkaProducer) Produce(topic, key string, value []byte) error {
	p.mu.Lock()
	defer p.mu.Unlock()
	p.messages = append(p.messages, struct {
		Topic string
		Key   string
		Value []byte
	}{Topic: topic, Key: key, Value: value})
	return nil
}

// GetMessages returns produced messages
func (p *MockKafkaProducer) GetMessages() []struct {
	Topic string
	Key   string
	Value []byte
} {
	p.mu.RLock()
	defer p.mu.RUnlock()
	return p.messages
}

File: tests/mocks/mock_services.py
"""Mock services for Python integration tests."""

import asyncio
import json
from datetime import datetime, timedelta
from typing import Any, Dict, List, Optional
from unittest.mock import AsyncMock, MagicMock
from uuid import uuid4

from fastapi import FastAPI
from fastapi.responses import JSONResponse
from pydantic import BaseModel
import uvicorn


# Mock Data Models
class MockAlert(BaseModel):
    """Mock alert model."""
    id: str
    title: str
    severity: str
    status: str
    source: str
    timestamp: datetime
    description: Optional[str] = None
    metadata: Optional[Dict[str, Any]] = None


class MockEvent(BaseModel):
    """Mock event model."""
    event_id: str
    timestamp: datetime
    event_type: str
    source: Optional[Dict[str, Any]] = None
    message: Optional[str] = None
    metadata: Optional[Dict[str, Any]] = None


class MockTriageResult(BaseModel):
    """Mock triage result."""
    alert_id: str
    priority: str
    classification: str
    confidence: float
    reasoning: str


class MockAgentExecution(BaseModel):
    """Mock agent execution."""
    task_id: str
    state: str
    steps: List[Dict[str, Any]]
    current_step: int
    final_result: Optional[Dict[str, Any]] = None
    error: Optional[str] = None


# Mock Data Store
class MockDataStore:
    """In-memory data store for mocks."""

    def __init__(self):
        self.alerts: Dict[str, MockAlert] = {}
        self.events: List[MockEvent] = []
        self.triage_results: Dict[str, MockTriageResult] = {}
        self.agent_executions: Dict[str, MockAgentExecution] = {}
        self._load_sample_data()

    def _load_sample_data(self):
        """Load sample data into the store."""
        # Sample alerts
        self.alerts["alert-001"] = MockAlert(
            id="alert-001",
            title="SSH Brute Force Detected",
            severity="high",
            status="new",
            source="detection_engine",
            timestamp=datetime.utcnow() - timedelta(hours=1),
            description="Multiple failed SSH login attempts detected",
        )

        self.alerts["alert-002"] = MockAlert(
            id="alert-002",
            title="Malware Detected",
            severity="critical",
            status="investigating",
            source="edr",
            timestamp=datetime.utcnow() - timedelta(minutes=30),
            description="Ransomware signature detected on endpoint",
        )

    def add_alert(self, alert: MockAlert) -> MockAlert:
        """Add an alert to the store."""
        if not alert.id:
            alert.id = f"alert-{uuid4()}"
        self.alerts[alert.id] = alert
        return alert

    def add_event(self, event: MockEvent) -> MockEvent:
        """Add an event to the store."""
        if not event.event_id:
            event.event_id = f"evt-{uuid4()}"
        self.events.append(event)
        return event


# Mock Triage Service
def create_mock_triage_app(store: MockDataStore) -> FastAPI:
    """Create a mock triage service FastAPI app."""
    app = FastAPI(title="Mock Triage Service")

    @app.get("/health")
    async def health():
        return {"status": "healthy", "service": "ai-triage"}

    @app.get("/ready")
    async def ready():
        return {"status": "ready"}

    @app.post("/api/v1/triage")
    async def triage_alert(request: Dict[str, Any]):
        alert = request.get("alert", {})
        alert_id = alert.get("id", str(uuid4()))

        result = MockTriageResult(
            alert_id=alert_id,
            priority="high" if alert.get("severity") in ["high", "critical"] else "medium",
            classification="true_positive",
            confidence=0.85,
            reasoning="Mock triage result for testing",
        )

        store.triage_results[alert_id] = result

        return {
            "success": True,
            "data": result.dict(),
            "request_id": request.get("request_id"),
        }

    @app.post("/api/v1/triage/batch")
    async def triage_batch(request: Dict[str, Any]):
        alerts = request.get("alerts", [])
        results = []

        for alert in alerts:
            alert_id = alert.get("id", str(uuid4()))
            result = MockTriageResult(
                alert_id=alert_id,
                priority="medium",
                classification="needs_review",
                confidence=0.75,
                reasoning="Batch mock triage",
            )
            results.append(result.dict())

        return {
            "success": True,
            "data": results,
            "request_id": request.get("request_id"),
        }

    @app.post("/api/v1/classify")
    async def classify_alert(request: Dict[str, Any]):
        return {
            "success": True,
            "data": {
                "category": "malware",
                "confidence": 0.9,
                "categories_scores": {
                    "malware": 0.9,
                    "phishing": 0.05,
                    "brute_force": 0.05,
                },
            },
        }

    @app.post("/api/v1/correlate")
    async def correlate_alerts(request: Dict[str, Any]):
        alerts = request.get("alerts", [])
        return {
            "success": True,
            "data": {
                "correlation_id": str(uuid4()),
                "correlated_alerts": [a.get("alert_id") for a in alerts],
                "correlation_type": "temporal",
                "confidence": 0.8,
            },
        }

    return app


# Mock Copilot Service
def create_mock_copilot_app(store: MockDataStore) -> FastAPI:
    """Create a mock copilot service FastAPI app."""
    app = FastAPI(title="Mock Copilot Service")

    @app.get("/health")
    async def health():
        return {"status": "healthy", "service": "ai-copilot"}

    @app.post("/api/v1/query/natural")
    async def nl_query(request: Dict[str, Any]):
        question = request.get("question", "")
        return {
            "success": True,
            "data": {
                "generated_sql": "SELECT count(*) FROM events WHERE timestamp >= now() - INTERVAL 1 HOUR",
                "explanation": f"Query generated for: {question}",
                "results": [],
            },
        }

    @app.post("/api/v1/investigate")
    async def investigate(request: Dict[str, Any]):
        alert = request.get("alert", {})
        return {
            "success": True,
            "data": {
                "alert_id": alert.get("id"),
                "findings": [
                    "Initial analysis suggests malicious activity",
                    "Source IP has no prior history",
                    "Similar patterns detected in recent alerts",
                ],
                "recommended_actions": [
                    "Isolate affected host",
                    "Collect forensic evidence",
                    "Block source IP at firewall",
                ],
                "confidence": 0.85,
            },
        }

    @app.post("/api/v1/recommend")
    async def recommend(request: Dict[str, Any]):
        return {
            "success": True,
            "data": {
                "recommendations": [
                    {
                        "action": "Isolate affected systems",
                        "priority": "high",
                        "rationale": "Prevent lateral movement",
                    },
                    {
                        "action": "Collect forensic evidence",
                        "priority": "high",
                        "rationale": "Preserve evidence for investigation",
                    },
                ],
                "playbook_suggestions": ["playbook-malware-response"],
            },
        }

    @app.post("/api/v1/summarize")
    async def summarize(request: Dict[str, Any]):
        incident = request.get("incident", {})
        return {
            "success": True,
            "data": {
                "summary": f"Incident {incident.get('incident_id')} involves potential malware activity.",
                "key_findings": ["Malware detected", "Lateral movement attempted"],
                "timeline_summary": "Attack progression over 2 hours",
            },
        }

    @app.post("/api/v1/chat")
    async def chat(request: Dict[str, Any]):
        message = request.get("message", "")
        return {
            "success": True,
            "data": {
                "response": f"Mock response to: {message}",
                "suggestions": ["Follow up question 1", "Follow up question 2"],
            },
        }

    return app


# Mock Agentic Service
def create_mock_agentic_app(store: MockDataStore) -> FastAPI:
    """Create a mock agentic service FastAPI app."""
    app = FastAPI(title="Mock Agentic Service")

    @app.get("/health")
    async def health():
        return {"status": "healthy", "service": "ai-agentic", "version": "0.1.0"}

    @app.post("/api/v1/execute")
    async def execute_task(request: Dict[str, Any]):
        task = request.get("task", {})
        task_id = task.get("task_id", str(uuid4()))

        execution = MockAgentExecution(
            task_id=task_id,
            state="pending",
            steps=[],
            current_step=0,
        )

        store.agent_executions[task_id] = execution

        return {
            "success": True,
            "data": execution.dict(),
            "request_id": request.get("request_id"),
        }

    @app.get("/api/v1/tasks/{task_id}")
    async def get_task(task_id: str):
        execution = store.agent_executions.get(task_id)
        if execution:
            return {
                "success": True,
                "data": execution.dict(),
            }
        return JSONResponse(
            status_code=404,
            content={"success": False, "error": "Task not found"},
        )

    @app.get("/api/v1/tasks")
    async def list_tasks():
        return {
            "success": True,
            "data": [e.dict() for e in store.agent_executions.values()],
        }

    @app.post("/api/v1/approve")
    async def approve_action(request: Dict[str, Any]):
        task_id = request.get("task_id")
        approved = request.get("approved", False)

        execution = store.agent_executions.get(task_id)
        if execution:
            execution.state = "running" if approved else "cancelled"
            return {
                "success": True,
                "data": execution.dict(),
            }

        return JSONResponse(
            status_code=404,
            content={"success": False, "error": "Task not found"},
        )

    @app.post("/api/v1/cancel/{task_id}")
    async def cancel_task(task_id: str):
        execution = store.agent_executions.get(task_id)
        if execution:
            execution.state = "cancelled"
            return {
                "success": True,
                "data": execution.dict(),
            }

        return JSONResponse(
            status_code=404,
            content={"success": False, "error": "Task not found"},
        )

    return app


# Mock HTTP Client for Testing
class MockHTTPClient:
    """Mock HTTP client for testing."""

    def __init__(self, responses: Optional[Dict[str, Any]] = None):
        self.responses = responses or {}
        self.requests: List[Dict[str, Any]] = []

    async def get(self, url: str, **kwargs) -> Dict[str, Any]:
        """Mock GET request."""
        self.requests.append({"method": "GET", "url": url, "kwargs": kwargs})
        return self._get_response(url, "GET")

    async def post(self, url: str, **kwargs) -> Dict[str, Any]:
        """Mock POST request."""
        self.requests.append({"method": "POST", "url": url, "kwargs": kwargs})
        return self._get_response(url, "POST")

    async def put(self, url: str, **kwargs) -> Dict[str, Any]:
        """Mock PUT request."""
        self.requests.append({"method": "PUT", "url": url, "kwargs": kwargs})
        return self._get_response(url, "PUT")

    async def delete(self, url: str, **kwargs) -> Dict[str, Any]:
        """Mock DELETE request."""
        self.requests.append({"method": "DELETE", "url": url, "kwargs": kwargs})
        return self._get_response(url, "DELETE")

    def _get_response(self, url: str, method: str) -> Dict[str, Any]:
        """Get mock response for URL and method."""
        key = f"{method}:{url}"
        if key in self.responses:
            return self.responses[key]

        # Default responses
        if "/health" in url:
            return {"status": 200, "body": {"status": "healthy"}}

        return {"status": 200, "body": {}}


# Mock ClickHouse Client
class MockClickHouseClient:
    """Mock ClickHouse client for testing."""

    def __init__(self):
        self.queries: List[str] = []
        self.results: Dict[str, Any] = {
            "SELECT count() FROM events": [{"count()": 1000}],
        }

    async def execute(self, query: str, params: Optional[Dict] = None) -> List[Dict]:
        """Execute a mock query."""
        self.queries.append(query)

        if query in self.results:
            return self.results[query]

        return []

    def set_result(self, query: str, result: Any):
        """Set mock result for a query."""
        self.results[query] = result


# Mock Kafka Producer
class MockKafkaProducer:
    """Mock Kafka producer for testing."""

    def __init__(self):
        self.messages: List[Dict[str, Any]] = []

    async def send(self, topic: str, value: bytes, key: Optional[bytes] = None):
        """Send a mock message."""
        self.messages.append({
            "topic": topic,
            "key": key,
            "value": value,
            "timestamp": datetime.utcnow().isoformat(),
        })

    async def flush(self):
        """Flush mock producer."""
        pass


# Utility functions for running mock servers
def run_mock_triage_server(port: int = 8001):
    """Run mock triage server."""
    store = MockDataStore()
    app = create_mock_triage_app(store)
    uvicorn.run(app, host="0.0.0.0", port=port)


def run_mock_copilot_server(port: int = 8000):
    """Run mock copilot server."""
    store = MockDataStore()
    app = create_mock_copilot_app(store)
    uvicorn.run(app, host="0.0.0.0", port=port)


def run_mock_agentic_server(port: int = 8002):
    """Run mock agentic server."""
    store = MockDataStore()
    app = create_mock_agentic_app(store)
    uvicorn.run(app, host="0.0.0.0", port=port)


if __name__ == "__main__":
    import sys

    if len(sys.argv) > 1:
        service = sys.argv[1]
        if service == "triage":
            run_mock_triage_server()
        elif service == "copilot":
            run_mock_copilot_server()
        elif service == "agentic":
            run_mock_agentic_server()
        else:
            print(f"Unknown service: {service}")
            print("Usage: python mock_services.py [triage|copilot|agentic]")
    else:
        print("Usage: python mock_services.py [triage|copilot|agentic]")
